HotelReservation/HotelReservation/StepExecutor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using HotelReservation.Hotel;
+using HotelReservation.Mail;
+using HotelReservation.Payment;
+
+namespace HotelReservation
+{
+    public class StepExecutor
+    {
+        private HotelSystem _hotelSystem;
+        private PaymentSystemReservationStepCreator _paymentSystemReservationStepCreator;
+        private MailSystemReservationStepCreator _mailSystemReservationStepCreator;
+
+        public StepExecutor(HotelSystem hotelSystem, PaymentSystemReservationStepCreator paymentSystemReservationStepCreator,
+            MailSystemReservationStepCreator mailSystemReservationStepCreator)
+        {
+            _hotelSystem = hotelSystem;
+            _paymentSystemReservationStepCreator = paymentSystemReservationStepCreator;
+            _mailSystemReservationStepCreator = mailSystemReservationStepCreator;
+        }
+
+        public int ExecuteStep(IConvertible reservationStep)
```

Comment:
Co ta klasa zwraca? Czasem 0, czasem hotel id (przynajmniej tego się spodziewasz tam, gdzie ją odpalasz). Czysty kontrakt to połowa sukcesu. Jeśli klasa zwraca hotelId to powinna to robić zawsze. Nie może być tak, że raz jakiś numer ma jedno znaczenie a raz innne. Musisz sobie odpowiedzieć na pytanie, co jest wynikiem stepu. Jeśli wynik ma kilka informacji - możesz zrobić obiekt. Weź pod uwagę, że ta klasa odpala jakieś abstrakcynje interfejsy. Ona nie wie, co dokładnie robią odpalane krotki. Powinna wiedzieć tylko, co charakteryzuje wszystkie stepy. Np. że krok może się zakończyć sukcesem, lub nie. Kroki nic nie zwracają i nie mają wyniku. Pojawił się problem, bo pomieszałeś dwa rodzaje kroków. Na obecną chwilę zwracany rezultat powinien być void, a obsługa konsoli wylatuje stąd

Reply:
Fixed - obsługa konsoli usunięta, ale myślę, że muszę dodać zwracanie jakiegoś obiektu jako wynik wywołania HotelSelect.SelectHotel()